### ISSUE TITLE
Should valid-version support `11.x`?

### ISSUE THREAD

**Issue by thernstig** [JoshuaKGoldberg/eslint-plugin-package-json#2091]
I had this in my package.json:

```json
  "devEngines": {
    "packageManager": {
      "name": "pnpm",
      "version": "11.x",
      "onFail": "error"
    },
    "runtime": {
      "name": "node",
      "version": "24.19.0",
      "onFail": "error"
    }
  }
```

It fails on `11.x`, but I do not want to write a bug as I am not even sure it is supported or not. Maybe it is a bug?

edit: It looks like it should be supported, so it is a bug? See https://www.npmjs.com/package/semver.

**Comment by michaelfaith** [comment]
`valid-version` shouldn't touch `packageManager.version` at all. It should only validate the top-level `version` field. Are you saying that's not the case?

**Comment by thernstig** [comment]
It touches `devEngines.packageManager.version` for sure. Changing to this fixes it:

```json
  "devEngines": {
    "packageManager": {
      "name": "pnpm",
      "version": "^11.0.0",
      "onFail": "error"
    },
    "runtime": {
      "name": "node",
      "version": "24.19.0",
      "onFail": "error"
    }
  }
```

**Comment by thernstig** [comment]
Just for more context, since this is indeed a bug then, I am using the latest version. The error looks like this:

```sh
> eslint

Oops! Something went wrong! :(

ESLint: 10.8.1

TypeError: Invalid Version: 11
Occurred while linting /home/userA/code/projectA/package.json:16
Rule: "package-json/valid-dependencies"
    at new SemVer (/home/userA/code/projectA/node_modules/.pnpm/semver@7.8.5/node_modules/semver/classes/semver.js:56:13)
    at compare (/home/userA/code/projectA/node_modules/.pnpm/semver@7.8.5/node_modules/semver/functions/compare.js:5:3)
    at Object.gte (/home/userA/code/projectA/node_modules/.pnpm/semver@7.8.5/node_modules/semver/functions/gte.js:4:30)
    at packageManagerAwareValidateDependencies (file:///home/userA/code/projectA/node_modules/.pnpm/eslint-plugin-package-json@1.7.0_@eslint+json@2.0.1_eslint@10.8.1_jiti@2.7.0_supports-color@10.2.2_/node_modules/eslint-plugin-package-json/dist/plugin-Dzfp9IK7.mjs:1737:26)
    at acc.<computed> (file:///home/userA/code/projectA/node_modules/.pnpm/eslint-plugin-package-json@1.7.0_@eslint+json@2.0.1_eslint@10.8.1_jiti@2.7.0_supports-color@10.2.2_/node_modules/eslint-plugin-package-json/dist/plugin-Dzfp9IK7.mjs:1595:22)
    at file:///home/userA/code/projectA/node_modules/.pnpm/eslint-json-compat-utils@0.2.3_@eslint+json@2.0.1_eslint@10.8.1_jiti@2.7.0_supports-col_699bf3f7c3c9c8b425272f547ee1ef8b/node_modules/eslint-json-compat-utils/dist/index.mjs:686:9
    at file:///home/userA/code/projectA/node_modules/.pnpm/eslint-json-compat-utils@0.2.3_@eslint+json@2.0.1_eslint@10.8.1_jiti@2.7.0_supports-col_699bf3f7c3c9c8b425272f547ee1ef8b/node_modules/eslint-json-compat-utils/dist/index.mjs:700:50
    at Array.forEach (<anonymous>)
    at file:///home/userA/code/projectA/node_modules/.pnpm/eslint-json-compat-utils@0.2.3_@eslint+json@2.0.1_eslint@10.8.1_jiti@2.7.0_supports-col_699bf3f7c3c9c8b425272f547ee1ef8b/node_modules/eslint-json-compat-utils/dist/index.mjs:700:34
    at invokeWithConvertedNode (file:///home/userA/code/projectA/node_modules/.pnpm/eslint-json-compat-utils@0.2.3_@eslint+json@2.0.1_eslint@10.8.1_jiti@2.7.0_supports-col_699bf3f7c3c9c8b425272f547ee1ef8b/node_modules/eslint-json-compat-utils/dist/index.mjs:716:7)
```

An AI described the problem (I have not verified this):

> This is a bug in the package-json ESLint plugin, not an actual invalid dependency in your manifest. The rule is enabled for [eslint.config.ts:233](vscode-file://vscode-app/c:/Users/siniz/AppData/Local/Programs/Microsoft%20VS%20Code/df53daabb1/resources/app/out/vs/code/electron-browser/workbench/workbench.html), and while "package-json/valid-dependencies" is checking your package file it also inspects [package.json:77](vscode-file://vscode-app/c:/Users/siniz/AppData/Local/Programs/Microsoft%20VS%20Code/df53daabb1/resources/app/out/vs/code/electron-browser/workbench/workbench.html), specifically [package.json:80](vscode-file://vscode-app/c:/Users/siniz/AppData/Local/Programs/Microsoft%20VS%20Code/df53daabb1/resources/app/out/vs/code/electron-browser/workbench/workbench.html). In the plugin code, [plugin-X4SuFW5t.mjs:1638](vscode-file://vscode-app/c:/Users/siniz/AppData/Local/Programs/Microsoft%20VS%20Code/df53daabb1/resources/app/out/vs/code/electron-browser/workbench/workbench.html) turns "11.x" into "11", and [plugin-X4SuFW5t.mjs:1668](vscode-file://vscode-app/c:/Users/siniz/AppData/Local/Programs/Microsoft%20VS%20Code/df53daabb1/resources/app/out/vs/code/electron-browser/workbench/workbench.html) passes that to semver.gte, which throws "Invalid Version: 11". I verified locally that semver accepts "11.x" as a range, and only rejects the plugin’s broken normalized value "11".

line 80 here refers to `"version": "11.x",` and line 70 is `devEngines`.

**Comment by michaelfaith** [comment]
That error message says `valid-dependencies`, not `valid-version`. Would you mind providing a minimum reproduction?

**Comment by michaelfaith** [comment]
Nevermind, i think i see what it is. This is a regression introduced by #2070. Does it work in 1.6.2?

**Comment by michaelfaith** [comment]
https://github.com/michaelfaith/eslint-plugin-package-json/pull/2092 

Thanks for reporting this!  In the future, please use our issue template when filing issues.

**Comment by thernstig** [comment]
Will do! I was unsure it was a bug, but suppose I should use the bug report template anyway for a potential bug.
